pepper-apis/src/main/java/org/broadinstitute/ddp/copy/AnswerToAnswerCopier.java

```diff
@@ -0,0 +1,176 @@
+                            targetParentAnswer.getAnswerId(),
+                            List.of(targetChild.getAnswerId()),
+                            List.of(rowIdx));
+                } else {
+                    targetChild = updateTargetAnswer(sourceChild, targetInstance.getId(), targetChild);
+                }
+                targetRow.set(colIdx, targetChild);
+            }
+        }
+    }
+
+    private void copySourceAnswer(Answer sourceAnswer, FormResponse targetInstance, QuestionDto targetQuestion) {
+        Answer targetAnswer = targetInstance.getAnswer(targetQuestion.getStableId());
+        if (targetAnswer == null) {
+            targetAnswer = createTargetAnswer(sourceAnswer, targetInstance.getId(), targetQuestion);
+        } else {
+            targetAnswer = updateTargetAnswer(sourceAnswer, targetInstance.getId(), targetAnswer);
+        }
+        targetInstance.putAnswer(targetAnswer);
+    }
+
+    private Answer createTargetAnswer(Answer sourceAnswer, long targetInstanceId, QuestionDto targetQuestion) {
+        Answer targetAnswer;
+        QuestionType type = sourceAnswer.getQuestionType();
+        if (type == QuestionType.AGREEMENT) {
+            boolean value = ((AgreementAnswer) sourceAnswer).getValue();
+            targetAnswer = new AgreementAnswer(null, targetQuestion.getStableId(), null, value);
+        } else if (type == QuestionType.BOOLEAN) {
+            boolean value = ((BoolAnswer) sourceAnswer).getValue();
+            targetAnswer = new BoolAnswer(null, targetQuestion.getStableId(), null, value);
+        } else if (type == QuestionType.COMPOSITE) {
+            throw new DDPException("Copying from top-level composite is not supported");
+        } else if (type == QuestionType.DATE) {
+            DateValue value = ((DateAnswer) sourceAnswer).getValue();
+            targetAnswer = new DateAnswer(null, targetQuestion.getStableId(), null, value);
+        } else if (type == QuestionType.NUMERIC) {
+            Long value = ((NumericIntegerAnswer) sourceAnswer).getValue();
+            targetAnswer = new NumericIntegerAnswer(null, targetQuestion.getStableId(), null, value);
+        } else if (type == QuestionType.PICKLIST) {
+            List<SelectedPicklistOption> value = ((PicklistAnswer) sourceAnswer).getValue();
+            targetAnswer = new PicklistAnswer(null, targetQuestion.getStableId(), null, value);
+        } else if (type == QuestionType.TEXT) {
+            String value = ((TextAnswer) sourceAnswer).getValue();
+            targetAnswer = new TextAnswer(null, targetQuestion.getStableId(), null, value);
+        } else {
+            throw new DDPException("Unhandled copying for answer type " + type);
+        }
+        return answerDao.createAnswer(operatorId, targetInstanceId, targetQuestion.getId(), targetAnswer);
+    }
+
+    private Answer updateTargetAnswer(Answer sourAnswer, long targetInstanceId, Answer targetAnswer) {
```

Comment:
`sourAnswer` indeed!  I would prefer this exception be thrown directly by client code that thinks it needs to issue an update.  It's possible that someone could see the method name, _not_ look at the implementation, and think "Oh cool, I'll just call `updateTargetAnswer` and then...surprise!

Reply:
Ahh typo! And sure, we can have it just throw an exception for now.